packages/react/src/components/Storage/FileUploader/Previewer/__tests__/Previewer.test.tsx

```diff
@@ -4,9 +4,81 @@ import { render } from '@testing-library/react';
 import { Previewer } from '..';
 
 describe('Previewer', () => {
+  let fakeFile;
+  let fakeFile2;
+  beforeEach(() => {
+    fakeFile = new File(['hello'], 'hello.png', { type: 'image/png' });
+    fakeFile2 = new File(['goodbye'], 'goodbye.png', {
+      type: 'image/png',
+    });
+  });
   it('exists', async () => {
-    const { container } = render(<Previewer />);
+    const { container } = render(
+      <Previewer
+        acceptedFileTypes={['.png']}
+        files={[fakeFile]}
+        inDropZone={false}
+        multiple={true}
+        onClose={() => ''}
+        onDragEnter={() => ''}
+        onDragLeave={() => ''}
+        onDragOver={() => ''}
+        onDragStart={() => ''}
+        onDrop={() => ''}
+        onFileChange={() => ''}
+        fileNames={[]}
+        level="public"
+      />
+    );
 
     expect(container).toBeTruthy();
   });
+  it('has two files passed in and displays them', async () => {
+    const { container, findByText } = render(
+      <Previewer
+        acceptedFileTypes={['.png']}
+        files={[fakeFile, fakeFile2]}
+        inDropZone={false}
+        multiple={true}
+        onClose={() => ''}
+        onDragEnter={() => ''}
+        onDragLeave={() => ''}
+        onDragOver={() => ''}
+        onDragStart={() => ''}
+        onDrop={() => ''}
+        onFileChange={() => ''}
+        fileNames={[]}
+        level="public"
+      />
+    );
+
+    expect(
+      container.querySelectorAll('.amplify-fileuploader-file').length
+    ).toEqual(2);
+
+    expect(await findByText('hello.png')).toBeVisible();
+    expect(await findByText('goodbye.png')).toBeVisible();
+  });
+
+  it('has one file passed in and shows one selected ', async () => {
```

Comment:
These unit tests only test for visual UI. Are there any functionalities, or unhappy cases that need to be added here?

Reply:
There is not too much functionality here yet, maybe a test regarding the drag and drop would help. I"m still debating whether that should be an e2e test. I don't see great reat-test-util utilities for drag and drop testing.

Reply:
That's fair, since this component is presentational. We'll keep an eye on testing as we make more PRs, thanks!